### ISSUE TITLE
WebViewClientFlutterApiImpl.requestLoading: java.lang.NullPointerException

### ISSUE THREAD

**Issue by GeylanKalafMohe** [flutter/flutter#123004]
## Steps to Reproduce

1. Run the WebView Widget on Android from [webview_flutter](https://pub.dev/packages/webview_flutter) on this specs:

Device
Brand:Samsung
Model:Galaxy A53 5G
Orientation: Portrait
RAM free: 2.14 GB
Disk free: 68.61 GB
Operating System
Version:Android 13
Orientation: Portrait
Rooted:No
Crash

**Expected results:**
Should load the page normally

**Actual results:**
Crash with this logs in Firebase Crashlytics:
```
Fatal Exception: java.lang.NullPointerException:
       at java.util.Objects.requireNonNull(Objects.java:220)
       at io.flutter.plugins.webviewflutter.WebViewClientFlutterApiImpl.requestLoading(WebViewClientFlutterApiImpl.java:13)
       at io.flutter.plugins.webviewflutter.WebViewClientHostApiImpl$WebViewClientImpl.shouldOverrideUrlLoading(WebViewClientHostApiImpl.java:4)
       at org.chromium.android_webview.AwContentsClientBridge.shouldOverrideUrlLoading(AwContentsClientBridge.java:89)
       at android.os.MessageQueue.nativePollOnce(MessageQueue.java)
       at android.os.MessageQueue.next(MessageQueue.java:335)
       at android.os.Looper.loopOnce(Looper.java:186)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8741)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```

<details>
<summary>Code sample</summary>

```dart
WebViewWidget(controller: WebViewController()..loadRequest(Uri.parse('https://www.google.com')))
```

</details>

<details>
  <summary>Logs</summary>

```
flutter doctor -v
[✓] Flutter (Channel stable, 3.7.7, on macOS 13.2.1 22D68 darwin-x64, locale en-SA)
    • Flutter version 3.7.7 on channel stable at /Users/geylankalafmohe/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2ad6cd72c0 (12 days ago), 2023-03-08 09:41:59 -0800
    • Engine revision 1837b5be5f
    • Dart version 2.19.4
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/geylankalafmohe/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.76.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.60.0

[✓] Connected device (3 available)
    • Dev iPad (mobile) • 00008101-000E40C80183001E • ios            • iOS 16.3.1 20D67
    • macOS (desktop)   • macos                     • darwin-x64     • macOS 13.2.1 22D68 darwin-x64
    • Chrome (web)      • chrome                    • web-javascript • Google Chrome 111.0.5563.64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by huycozy** [comment]
Hi @GeylanKalafMohe 
Which plugin version are you using? 

Also, it looks like the issue only happens in production, right? Can you reproduce this locally in release mode? (I checked this on the latest plugin version in debug mode but didn't see the crash)

**Comment by GeylanKalafMohe** [comment]
> Hi @GeylanKalafMohe Which plugin version are you using?
> 
> Also, it looks like the issue only happens in production, right? Can you reproduce this locally in release mode? (I checked this on the latest plugin version in debug mode but didn't see the crash)

  webview_flutter: ^4.0.6
  
I also tested on the same Android version in DebugMode however it does not seem to be crashing. I received that crash in Production Environment.

**Comment by GeylanKalafMohe** [comment]
![image](https://user-images.githubusercontent.com/42656134/226602171-1847e198-fad0-4698-a4d8-1739b54b2588.png)

All these issues are appearing for webview_flutter on Android

**Comment by GeylanKalafMohe** [comment]
@huycozy I can not reproduce any of the crashes i got from this package with my current setup and with simple setup. But still it is happening on production and like what the crashes states, it is coming from the package itself.

There was an issue posted already before which mentioned that they tackled and fixed this but it still happens.

Please check on the package why and when it is throwing the exception.

**Comment by huycozy** [comment]
Thanks for the response. I think the issue you are referring to is https://github.com/flutter/flutter/issues/119899, though it had the different crash message (`Could not find identifier for WebView`).  

Sometimes, the NullPointerException can happen in Release mode but not in Debug mode. Once it can be reproducible in debug mode, it will be helpful for investigating the issue. That is the reason why need to check this in release mode.

Labeling the issue for further investigation.

**Comment by bparrishMines** [comment]
@GeylanKalafMohe Is your app a pure Flutter app or are you using [add-to-app](https://docs.flutter.dev/development/add-to-app).

**Comment by GeylanKalafMohe** [comment]
@bparrishMines Pure Flutter app

**Comment by bparrishMines** [comment]
This is most likely caused by a race condition where the plugin is detached from the engine which closes the  `InstanceManager`. Meanwhile callbacks from the `WebView` are still occurring as the app is closing.

**Comment by GeylanKalafMohe** [comment]
@bparrishMines Could be.. Can this be tackled in a new package update fast?

**Comment by GeylanKalafMohe** [comment]
?

**Comment by bparrishMines** [comment]
This should be fixed by https://github.com/flutter/packages/pull/3571

**Comment by GeylanKalafMohe** [comment]
@bparrishMines Great. Please let us know when it's available for a package update.

**Comment by ambogatyrev** [comment]
Im also facing this issue. Any update?

**Comment by bparrishMines** [comment]
This should be fixed by https://github.com/flutter/packages/pull/3571 and released with `webview_flutter_android: 3.5.2`. You can use `flutter pub upgrade` to get the latest version.
